apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -3228,10 +3229,48 @@ public static Bundle createBundle(
    *     Patient}s, which may contain multiple matching resources, or may also be empty.
    */
   public static Bundle addResourcesToBundle(Bundle bundle, List<IBaseResource> resources) {
+    Set<String> beneIds = new HashSet<String>();
     for (IBaseResource res : resources) {
       BundleEntryComponent entry = bundle.addEntry();
       entry.setResource((Resource) res);
+
+      if (entry.getResource().getResourceType().toString() == "ExplanationOfBenefit") {
+        ExplanationOfBenefit eob = ((ExplanationOfBenefit) entry.getResource());
+        if (eob != null
+            && eob.getPatient() != null
+            && !Strings.isNullOrEmpty(eob.getPatient().getReference())) {
+          String reference = eob.getPatient().getReference().replace("Patient/", "");
+          if (!Strings.isNullOrEmpty(reference)) {
+            beneIds.add(reference);
+          }
+        }
+      }
+
+      if (entry.getResource().getResourceType().toString() == "Patient") {
+        Patient patient = ((Patient) entry.getResource());
+        if (patient != null && !Strings.isNullOrEmpty(patient.getId())) {
+          beneIds.add(patient.getId());
+          ;
+        }
+      }
+
+      if (entry.getResource().getResourceType().toString() == "Coverage") {
+        Coverage coverage = ((Coverage) entry.getResource());
+        if (coverage != null
+            && coverage.getBeneficiary() != null
+            && !Strings.isNullOrEmpty(coverage.getBeneficiary().getReference())) {
+          String reference = coverage.getBeneficiary().getReference().replace("Patient/", "");
+          if (!Strings.isNullOrEmpty(reference)) {
+            beneIds.add(reference);
+          }
+        }
+      }
```

Comment:
do we have a utility class that this logic could live in so that we don't need a separate copy in v1 and v2?